unlock-app/src/utils/checkoutActions.ts

```diff
@@ -33,3 +37,13 @@ export const setPurchasingLockAddress = (
   kind: 'setPurchasingLockAddress',
   address,
 })
+
+interface SetTransactionHash {
+  kind: 'setTransactionHash'
```

Comment:
It looks like an artifact of yestercode, but with definition of the type why do we also have the kind property of with a string value of its name?

Reply:
It's an implicit type predicate - each of these interfaces has a unique "kind;" this allows the compiler to infer the following:

```javascript
if (thing.kind = "setTransactionHash") {
  // thing is now understood to be of the type SetTransactionHash
}
```

This adds some help while editing (you'll get autocomplete for the relevant properties) and it also ensures you won't try to access something that isn't there.